cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1049 @@
+    - imperfectly annotated the location/edges of a bounding box.
+    Any of these annotation errors should lead to an image with a lower label quality score.
+
+    Score is between 0 and 1.
+
+        - 1 - clean label (given label is likely correct).
+        - 0 - dirty label (given label is likely incorrect).
+
+    A score is calculated for each of ``N`` images, with ``K`` total classes in the data.
+    Each image has ``L`` annotated bounding boxes and ``M`` predicted bounding boxes.
+
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image in the format
+       ``{'bboxes': np.ndarray((M,4)), 'labels': np.ndarray((M,)), 'image_name': str}`` where ``L`` is the number of annotated bounding boxes
+       for the `i`-th image and ``bboxes[j]`` is in the format ``[x,y,x,y]`` with given label ``labels[j]``. (``image_name`` is optional here)
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image
+        in the format ``np.ndarray((K,))`` and ``predictions[i][k]`` is of shape ``np.ndarray(M,5)``
+        where ``M`` is the number of predicted bounding boxes for class ``k`` and the five columns correspond to ``[x,y,x,y,pred_prob]`` returned
+        by the model.
+
+    method:
+        The method used to calculate label_quality_scores. Options:
+
+        - ``subtype_lqs``: calculates image score as a composite score of the quality of badly located, swapped and missing bounding boxes.
+
+    probability_threshold:
+        Bounding boxes in ``predictions`` with ``pred_prob`` below the threshold are not considered for computing `label_quality_scores`.
+        If you know what probability-threshold was used when producing predicted boxes from your trained object detector,
+        please supply the value that was used here. If not provided, this value is inferred based on the smallest observed
+        predicted probability for any of the predicted boxes.
+
+    verbose : bool, default = True
+      Set to ``False`` to suppress all print statements.
+
+    Returns
+    ---------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate images that are more likely mislabeled.
+    """
+
+    _assert_valid_inputs(
+        labels=labels,
+        predictions=predictions,
+        method=method,
+        threshold=probability_threshold,
+    )
+
+    return _compute_label_quality_scores(
+        labels=labels,
+        predictions=predictions,
+        method=method,
+        threshold=probability_threshold,
+        verbose=verbose,
+    )
+
+
+def issues_from_scores(label_quality_scores: np.ndarray, *, threshold: float = 0.1) -> np.ndarray:
+    """Returns a list of indices images with issues sorted from most to least severe.
```

Comment:
this language needs to be made more clear, and state why this method might be usefule